tx_test.go

```diff
@@ -18,6 +22,100 @@ import (
 	"go.etcd.io/bbolt/internal/btesting"
 )
 
+func TestTx_Check_Page(t *testing.T) {
+	bucketKey := "testBucket"
+	db := btesting.MustCreateDBWithOption(t, &bolt.Options{PageSize: pageSize})
+	defer db.MustDeleteFile()
+
+	// arrange
+	if err := db.Update(func(tx *bolt.Tx) error {
+		b, err := tx.CreateBucketIfNotExists([]byte(bucketKey))
+		if err != nil {
+			t.Fatal(err)
+		}
+		// insert random data
+		for i := 0; i < 10; i++ {
+			insertRandKeysValuesBucket(t, b, 100)
+		}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+	db.MustCheck()
+
+	// act : corrupt a leaf page
+	corruptLeafPageFromDB(t, db.Path())
+
+	// assert: expected an error after check
+	if err := db.View(func(tx *bolt.Tx) error {
+		for err := range tx.Check() {
+			require.ErrorContains(t, err, "has unexpected type/flags: 20")
+		}
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+	db.MustClose()
+}
+
+func corruptLeafPageFromDB(t testing.TB, dbPath string) {
+	_, hwm, err := guts_cli.ReadPageAndHWMSize(dbPath)
+	if err != nil {
+		t.Fatalf("error reading HWM from dbPath %v: %v", dbPath, err)
+	}
+
+	var victimPage *common.Page
+	var victimBuf []byte
+
+	// find first leaf page starting page after root bucket's page
+	for pageID := uint64(4); pageID < uint64(hwm); {
+		p, buf, err := guts_cli.ReadPage(dbPath, pageID)
+		if err != nil {
+			t.Fatalf("error reading page %v from dbPath %v: %v", pageID, dbPath, err)
+		}
+
+		if p.IsLeafPage() {
+			victimPage = p
+			victimBuf = buf
+			break
+		}
+	}
+
+	if victimPage == nil || victimBuf == nil {
+		t.Fatalf("no leaf page is found in dbPath %v", dbPath)
+	}
+
+	// corrupt page's type
+	victimPage.SetFlags(0x20)
+	// write page back to file
+	err = guts_cli.WritePage(dbPath, victimBuf)
+	if err != nil {
+		t.Errorf("error writing victim page %v back to dbpath %v: %v", victimPage.Id(), dbPath, err)
+	}
+}
+
+func insertRandKeysValuesBucket(t testing.TB, bk *bolt.Bucket, n int) {
```

Comment:
@ahrtr this is being used in https://github.com/etcd-io/bbolt/pull/635 

once it is merged we can reuse and remove from here